File: ArtFlow/ArtFlow.Core/Entities/Order.cs
using ArtFlow.Core.Abstractions;
using ArtFlow.Core.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ArtFlow.Core.Entities
{
    public class Order : IBaseEntity
    {
        public int OrderId { get; set; }

        public string SellerId { get; set; }

        public User Seller { get; set; }

        public string CustomerId { get; set; }

        public User Customer { get; set; }

        public string? DriverId { get; set; }

        public User? Driver { get; set; }

        public int ArtpieceId { get; set; }

        public Artpiece Artpiece { get; set; }

        public int ExhibitionId { get; set; }

        public Exhibition Exhibition { get; set; }

        public string Adress { get; set; }

        public DeliveryStatus Status { get; set; }

        public DateTimeOffset UpdatedOn { get; set; } 

        public List<State> States { get; set; } = new List<State>();

    }
}

File: client-app/src/app/models/order.ts
import { Artpiece } from "./artpiece";
import { Exhibition } from "./exhibition";
import { State } from "./state";
import { User } from "./user";

export interface Order {
    orderId: number;
    seller: User;
    customer: User;
    driver: User | null;
    artpiece: Artpiece;
    exhibition: Exhibition;
    adress: string;
    status: number;
    updatedOn: Date;
    isStateOk: boolean;
    latestState: State;
}

export class OrderFormValues {
    artpieceId: string;
    exhibitionId: number;
    adress: string;

    constructor(artpieceId: string,
        exibitionId: number,
        adress: string){
            this.artpieceId = artpieceId;
            this.exhibitionId = exibitionId,
            this.adress = adress;
        }
}